pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java

```diff
@@ -48,15 +48,12 @@ public static void setExecutionStatistics(BaseResultsBlock resultsBlock, List<Op
       if (executionStatistics.getNumDocsScanned() > 0) {
         numSegmentsMatched++;
       }
-      // TODO: Check all operators and properly implement the getIndexSegment and remove this exception handling
```

Comment:
Don't remove this TODO. This TODO is the correct thing to do, and we should ensure all the top level operator (e.g. Select, Aggregation, GroupBy) has this method implemented.